Core/src/Detector/SupportBuilder.cpp

```diff
@@ -0,0 +1,213 @@
+      Acts::Vector3 planeAxisZ = {cosPhi, sinPhi, 0.};
+      Acts::Vector3 planeAxisY{0., 0., 1.};
+      Acts::Vector3 planeAxisX = planeAxisY.cross(planeAxisZ);
+
+      RotationMatrix3 planeRotation;
+      planeRotation.col(0) = planeAxisX;
+      planeRotation.col(1) = planeAxisY;
+      planeRotation.col(2) = planeAxisZ;
+
+      Transform3 sTransform{planeRotation};
+      sTransform.pretranslate(planeCenter);
+      // Place it
+      cSupport.push_back(
+          Surface::makeShared<PlaneSurface>(sTransform, sRectangle));
+    }
+  }
+
+  return cSupport;
+}
+
+std::vector<std::shared_ptr<Acts::Surface>>
+Acts::Experimental::SupportBuilder::discSupport(
+    const Transform3& transform, const std::array<ActsScalar, 4u>& bounds,
+    unsigned int splits) {
+  // Return vector
+  std::vector<std::shared_ptr<Acts::Surface>> dSupport;
+  if (splits == 1u) {
+    // No splitting is done in this case
+    dSupport.push_back(Surface::makeShared<DiscSurface>(
+        transform, std::make_shared<RadialBounds>(bounds)));
+  } else {
+    // Split into n(splits) planar surfaces in phi, prep work:
+    ActsScalar minR = bounds[0u];
+    ActsScalar maxR = bounds[1u];
+    ActsScalar minPhi = bounds[3u] - bounds[2u];
+    ActsScalar maxPhi = bounds[3u] + bounds[2u];
+    ActsScalar dHalfPhi = (maxPhi - minPhi) / (2 * splits);
+    ActsScalar cosPhiHalf = std::cos(dHalfPhi);
+    ActsScalar sinPhiHalf = std::sin(dHalfPhi);
+    ActsScalar maxLocY = maxR * cosPhiHalf;
+    ActsScalar minLocY = minR * cosPhiHalf;
+    ActsScalar hR = 0.5 * (maxLocY + minLocY);
+    ActsScalar hY = 0.5 * (maxLocY - minLocY);
+    ActsScalar hXminY = minR * sinPhiHalf;
+    ActsScalar hXmaxY = maxR * sinPhiHalf;
+    // Split trapezoid
+    auto sTrapezoid =
+        std::make_shared<Acts::TrapezoidBounds>(hXminY, hXmaxY, hY);
+    Vector3 zAxis = transform.rotation().col(2);
+    ActsScalar zPosition = transform.translation().z();
+    // Now create the Trapezoids
+    for (unsigned int iphi = 0; iphi < splits; ++iphi) {
+      // Create the split module transform
+      ActsScalar phi = -M_PI + (iphi + 0.5) * 2 * dHalfPhi;
+      auto sTransform = Transform3(
+          Translation3(hR * std::cos(phi), hR * std::sin(phi), zPosition) *
+          AngleAxis3(phi - 0.5 * M_PI, zAxis));
+      // Place it
+      dSupport.push_back(
+          Surface::makeShared<PlaneSurface>(sTransform, sTrapezoid));
+    }
+  }
+  return dSupport;
+}
+
+void Acts::Experimental::SupportBuilder::addSupport(
+    std::vector<std::shared_ptr<Surface>>& layerSurfaces,
+    std::vector<size_t>& assignToAll, const Extent& layerExtent,
+    Surface::SurfaceType layerRepresentation,
+    const std::array<ActsScalar, 5u>& layerSupportValues,
+    std::optional<Transform3> layerTransform, unsigned int supportSplits) {
+  // Cylinder and Disc section
+  if (layerRepresentation == Surface::SurfaceType::Cylinder or
+      layerRepresentation == Surface::SurfaceType::Disc) {
```

Comment:
looks like we only handle those two cases right now. should we error in the other cases?

Reply:
Correct. to be added later. I added an exception so far.